### PR TITLE
✨ colleciton by Id in

### LINKED ISSUE #48: `collectionListByIdIn`
### Context 

I want to fetch array of particular items, let's say 4 
instead of doing promise all on `client.getCollectionById` I can call `client. collectionListByIdIn`

### PR DESCRIPTION
**Pull request by roiLeo**
route is ugly but it's working

- close #48 
- fix failing test
```log
stdout | test/path.test.ts > Path utils > collecion id in > should do something
{
  baseURL: 'https://kodadot.squids.live/basick/graphql',
  query: {
    query: 'query  { collections: collectionEntities(where: {id_in: ["0xd9a2c93ba2e9fae10fe762a42ee807bbf95764cc","0x2f36072129aabb22bd4afffca0640a3e02695925"]})  { id, createdAt, name, image, metadata, currentOwner, issuer } }',
    variables: {}
  },
  path: '/base/collectionByIdIn/0xd9a2c93ba2e9fae10fe762a42ee807bbf95764cc,0x2f36072129aabb22bd4afffca0640a3e02695925'
}
```

```json
{
  "data": {
    "collections": [
      {
        "id": "0xd9a2c93ba2e9fae10fe762a42ee807bbf95764cc",
        "createdAt": "2024-05-20T17:02:17.000000Z",
        "name": "Basescapes",
        "image": "ipfs://bafybeiaystlm6xszy3eq5ovqgqjwxwgpeztdrvbsixiqx6me56yxp7efrq",
        "metadata": "ipfs://bafkreicljf6qmoula5u3l4jgrdilias6ucd54cecp7nw4iq2zeoesfee3e",
        "currentOwner": "0xE844b2a0a6453250c920BD2b4B7741946aB16C08",
        "issuer": "0xe844b2a0a6453250c920bd2b4b7741946ab16c08"
      },
      {
        "id": "0x2f36072129aabb22bd4afffca0640a3e02695925",
        "createdAt": "2024-09-10T08:30:31.000000Z",
        "name": "Harmony",
        "image": "ipfs://QmUyoKGVTiDTrQzjLdwj41q9K3rfHa7K68zQuoPXhnEm28",
        "metadata": "ipfs://QmcUD2wSowiUz9rxrSXgAbtiUWuZcoYGS7brJC359BCPw7",
        "currentOwner": "0xE844b2a0a6453250c920BD2b4B7741946aB16C08",
        "issuer": "0xe844b2a0a6453250c920bd2b4b7741946ab16c08"
      }
    ]
  }
}
```

// TODO:
- send id as query param
- `id: string[]`

### CODE REVIEW

File: src/clients/SquidClient.ts [hunk 1]
```diff
@@ -40,6 +40,15 @@ class SquidClient implements AbstractClient<SquidCollection, SquidNFT> {
     })
   }
 
+  collectionByIdIn(id: string, fields?: ObjProp<SquidCollection>): GraphQuery {
```
Comment (vikiival) [inline src/clients/SquidClient.ts:43]:
```suggestion
  collectionByIdIn(id: string | string[], fields?: ObjProp<SquidCollection>): GraphQuery {
```

File: src/clients/abstractClient.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { BaseEvent, GraphLike, GraphQuery, ObjProp, QueryProps } from '../types'
 // Collection, Token
 interface AbstractClient<C, T, E = BaseEvent> {
   collectionById(id: string, fields?: ObjProp<C>): GraphQuery
+  collectionByIdIn(id: string, fields?: ObjProp<C>): GraphQuery
```
Comment (vikiival) [inline src/clients/abstractClient.ts:5]:
```suggestion
  collectionByIdIn(id: string | string[], fields?: ObjProp<C>): GraphQuery
```

**Review by vikiival — COMMENTED** [review]
small

**Comment by roiLeo** [comment]
> small

> // TODO:
> 
>     * send id as query param

any idea how to achieve that?

**Comment by vikiival** [comment]
> any idea how to achieve that?

I would like to rewrite `/rest` package ideally :)

File: src/clients/SquidClient.ts [hunk 3]
```diff
@@ -40,6 +40,15 @@ class SquidClient implements AbstractClient<SquidCollection, SquidNFT> {
     })
   }
 
+  collectionByIdIn(id: string | string[], fields?: ObjProp<SquidCollection>): GraphQuery {
+    const toQuery = getFields(fields)
+    const list = JSON.stringify(id.split(','))
```
Comment (vikiival) [inline src/clients/SquidClient.ts:45]:
```suggestion
    const list = Array.isArray(id) ? id : JSON.stringify(id.split(','))
```

**Review by vikiival — COMMENTED** [review]
prob change impl

**Comment by vikiival** [comment]
Also pls merge main

**Comment by vikiival** [comment]
Works

```js
import { getClient }  from './dist/index.mjs'
const client = getClient('ahp')
const query = client.collectionByIdIn('1,13,163,171')
const result = await client.fetch(query)
console.log(result)
```

Does not work

```js
import { getClient }  from './dist/index.mjs'
const client = getClient('ahp')
const query = client.collectionByIdIn(['1', '13', '163', '171'])
const result = await client.fetch(query)
console.log(result)
```

cc @roiLeo one more fix plz

**Comment by roiLeo** [comment]
Should be ok, added more test
Capture d’écran 2024-10-09 à 4 07 25 PM

File: test/index.test.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ describe('UNIQUERY UTILS', () => {
 
     it('should return default kusama indexer', () => {
       const url = getUrl('ksm')
-      expect(url).eq('https://ahk.gql.api.kodadot.xyz/')
+      expect(url).eq('https://ksm.gql.api.kodadot.xyz/');
```
Comment (roiLeo) [inline test/index.test.ts:16]:
```suggestion
      expect(url).eq('https://ksm.gql.api.kodadot.xyz/')
```

**Pull request merged**